### DISCUSSION TITLE
How does linguist assign colors to the text mate scopes provided by the grammars?

### DISCUSSION THREAD

**Discussion by Animeshz** [github-linguist/linguist#5204] [Q&A] [answered]
Hi, I wanted to know how does linguist assign the colors to the respective scopes after the code is being parsed by the grammars. I did not found anything related to that in this repository I guess (or maybe I didn't explored it carefully :O, correct me if I'm wrong).

The question is raised because Kotlin's grammer [nishtahir/language-kotlin](https://github.com/nishtahir/language-kotlin) does add the annotations scope `meta.annotation.kotlin` but [is currently not given any treatment](https://github.com/github/linguist/issues/5085#issuecomment-724118549) by the github's syntax highlighter.

Is it managed somewhere closed-source, like the [github-lightshow](https://github-lightshow.herokuapp.com)? Or something else, I'm just not able to find.

**Comment by lildude** [comment, marked as answer]
Linguist doesn't assign the colours. This is entirely controlled by the stylesheet at https://github.com/primer/github-syntax-light (Lightshow uses this one) and https://github.com/primer/github-syntax-dark based on the grammar scopes applied by the grammar. @Alhadis provided a great summary of the grammar scopes to colour in his comment at https://github.com/github/linguist/pull/4568#issuecomment-513739638

**Reply by Alhadis** [reply]
[Here's a much better summary](https://github.com/Alhadis/language-etc/blob/master/samples/lists/scope-previews.nanorc). 😉
